utils/serviceRequest.js

```diff
@@ -62,59 +63,118 @@ class ABServiceRequest extends ServiceCote {
-               requestCount < attempts
-            ) {
-               this.req.log(
-                  `... timeout waiting for request (${key}), retrying ${requestCount}/${attempts}`
-               );
-               return await sendRequest();
-            } else {
-               throw err;
-            }
-         }
-      };
-      try {
-         const results = await sendRequest();
-         if (this.req.performance) {
-            this.req.performance.measure(key, key);
-         }
-         if (callback) return callback(null, results);
-         return results;
-      } catch (err) {
-         err._serviceRequest = key;
-         err._params = paramStack;
-         if (key != "log_manager.notification") {
-            // If there was an error requesting log_manger.notification, don't use req.notify or we can get caught in an infinite loop
-            this.req.notify.developer(err, {
-               message: `Could not request (${key}) - ${JSON.stringify(
-                  paramStack
-               )}`,
-            });
-         }
-         if (callback) return callback(err);
-         throw err;
-      }
+
+      var timeoutCleanup = false;
+      // {bool}
+      // are we attempting to clean up a timedout service call?
+
+      let returnPromise = new Promise((resolve, reject) => {
+         const sendRequest = () => {
+            countRequest += 1;
+
+            requester.send(
+               {
+                  ...paramStack,
+                  __timeout: timeout,
+               },
+               (err, results) => {
+                  let finalTime;
+                  if (this.req.performance) {
+                     finalTime = this.req.performance.measure(key, key);
+                  }
+
+                  if (err) {
+                     // https://github.com/dashersw/cote/blob/master/src/components/requester.js#L132
+                     if (err.message === "Request timed out.") {
+                        // Retry .send
+                        if (!timeoutCleanup && countRequest < attempts) {
+                           this.req.log(
+                              `... timeout waiting for request (${key}), retrying ${countRequest}/${ATTEMPT_REQUEST_MAXIMUM}`
+                           );
+
+                           sendRequest();
+                           return;
+                        }
+
+                        if (timeoutCleanup && countRequest < attempts) {
+                           // Q: should we attempt to scale our timeouts?
+                           timeout *= 1.5;
+
+                           this.req.log(
+                              `... OVERTIME: waiting for eventual response (${key}), retrying ${countRequest}/${ATTEMPT_REQUEST_OVERTIME}`
+                           );
+
+                           sendRequest();
+                           return;
+                        }
+
+                        this.req.notify.developer(err, {
```

Comment:
Since `req.notify()` uses `serviceRequest.request()` to pass the notifications to `log_manger.notification`. There is a potential to get caught in a loop if `log_manager` is down.

I'm not sure the best way to handle it, but I previously had the call to `notify` wrapped in `if (key != "log_manager.notification")`